tests/apollo/util/bft.py

```diff
@@ -431,6 +506,62 @@ def start_replica(self, replica_id):
                                             stdout=stdout_file,
                                             stderr=stderr_file,
                                             close_fds=True)
+            if notify_clients:
+                await self._notify_clients_replicas_started(self.all_client_ids(), [replica_id])
+
+    def replica_id_from_pid(self, pid):
+        """ Return an already-started replica id, according to a given pid """
+        for rep_id in self.procs.keys():
+            if self.procs[rep_id].pid == pid:
+                return rep_id
+        return None
+
+    def is_replica_id(self, node_id):
+        """ Return True if a node_id is a replica id (includes a read-only replica) """
+        return node_id < self.num_total_replicas()
+
+    def is_client_id(self, node_id):
+        """ Return True if node_id is a client Id - reserved or non-reserved """
+        start_id = self.num_total_replicas()
+        return start_id <= node_id < (start_id + + self.config.num_clients + RESERVED_CLIENTS_QUOTA)
```

Comment:
`+ +` ? Python doesn't seem to complain, but I'm wondering how does he interpret it 🐍

Reply:
This is a typo for sure, no idea how it works?
I tried "1 + + 1" on the interpreter, and it works....weird
fixed..